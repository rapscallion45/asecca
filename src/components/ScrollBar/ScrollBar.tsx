import { FC, ReactNode } from 'react';
import SimpleBarReact from 'simplebar-react';
import { alpha, styled } from '@mui/material/styles';

/**
 * Scrollbar Root Style
 *
 * Global application scrollbar root wrapper styling
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @component
 * @return {Component} - styled scrollbar wrapper component
 */
const RootStyle = styled('div')({
  flexGrow: 1,
  height: '100%',
  overflow: 'hidden',
});

/**
 * Scrollbar Style
 *
 * Global application scrollbar styling
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @component
 * @return {Component} - styled scrollbar component
 */
const SimpleBarStyle = styled(SimpleBarReact)(({ theme }) => ({
  maxHeight: '100%',
  '& .simplebar-scrollbar': {
    '&:before': {
      backgroundColor: alpha(theme.palette.grey[600], 0.48),
    },
    '&.simplebar-visible:before': {
      opacity: 1,
    },
  },
  '& .simplebar-track.simplebar-vertical': {
    width: 10,
  },
  '& .simplebar-track.simplebar-horizontal .simplebar-scrollbar': {
    height: 6,
  },
  '& .simplebar-mask': {
    zIndex: 'inherit',
  },
}));

/**
 * Scrollbar Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @typedef IScrollBarProps
 * @prop {ReactNode} children - component children nodes
 * @prop {any} sx - component styling options
 * @prop {any} other - component props to passed through to children
 */
interface IScrollBarProps {
  children: ReactNode;
  sx?: any;
  other?: any;
}

/**
 * Scrollbar
 *
 * Global application custom scrollbar
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @component
 * @param {IScrollBarProps} props - component props
 * @returns {FC} - scrollbar functional component
 */
const ScrollBar: FC<IScrollBarProps> = (props) => {
  const { children, sx, other } = props;

  return (
    <RootStyle>
      <SimpleBarStyle timeout={500} clickOnTrack={false} sx={sx} {...other}>
        {children}
      </SimpleBarStyle>
    </RootStyle>
  );
};

export default ScrollBar;
