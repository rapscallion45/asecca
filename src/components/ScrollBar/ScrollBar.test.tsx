import React, { FC } from 'react';
import renderer from 'react-test-renderer';
import ScrollBar from './ScrollBar';

/* test child component */
const TestChild: FC = () => <div>Test child</div>;

/**
 * Scrollbar Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 */
describe('ScrollBar', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <ScrollBar>
          <TestChild />
        </ScrollBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
