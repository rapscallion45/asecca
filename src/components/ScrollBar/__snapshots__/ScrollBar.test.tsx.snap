// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollBar Renders correctly 1`] = `
<div
  data-simplebar="init"
>
  <div
    className="simplebar-wrapper"
  >
    <div
      className="simplebar-height-auto-observer-wrapper"
    >
      <div
        className="simplebar-height-auto-observer"
      />
    </div>
    <div
      className="simplebar-mask"
    >
      <div
        className="simplebar-offset"
      >
        <div
          aria-label="scrollable content"
          className="simplebar-content-wrapper"
          role="region"
          tabIndex={0}
        >
          <div
            className="simplebar-content"
          >
            <div>
              Test child
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="simplebar-placeholder"
    />
  </div>
  <div
    className="simplebar-track simplebar-horizontal"
  >
    <div
      className="simplebar-scrollbar"
    />
  </div>
  <div
    className="simplebar-track simplebar-vertical"
  >
    <div
      className="simplebar-scrollbar"
    />
  </div>
</div>
`;
